src/osd/OSD.cc

```diff
@@ -9909,11 +9972,9 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb)
   osd->service.maybe_inject_dispatch_delay();
 
   // [lookup +] lock pg (if we have it)
-  if (!pg) {
-    pg = osd->_lookup_lock_pg(item.first);
-  } else {
-    pg->lock();
-  }
+  PGQueueable::OrderLocker::Ref l;
+  l = item.second.get_order_locker(pg, item.first, osd);
+  l->lock();
 
```

Comment:
@liu-chunmei you can refer #11188. (the pg lock or the op lock (or whatever) can be used if we created generalized interface)

@tchaikov I am also implementing the pg_slot part. How can we divide these works ?